### PR TITLE
Improve logging x1000

### PR DESCRIPTION
**Pull request by timja**
* Add timestamp
* Log parameters
* Fix unhandled promise when github isn't found
* Improve log messages
* Upgrade node.js

Fixes https://github.com/jenkins-infra/incrementals-publisher/issues/11

cc @basil

<details>
<summary>New log set with no github</summary>

```
2021-02-04T08:38:19.660Z info: 	Retrieving metadata from https://ci.jenkins.io/job/Stapler/job/stapler/job/PR-209/1/api/json?tree=actions[revision[hash,pullHash]]
2021-02-04T08:38:20.649Z info: 	Metadata loaded repo: stapler/stapler hash: e6cc839e83b65e27a0a860a44ae30f71fcbfb518
2021-02-04T08:38:20.650Z info: 	Prepared a temp dir for the archive /var/folders/7p/92qym3cj4p9bzr330tcx2m9w0000gn/T/incrementals-dQsme2
2021-02-04T08:38:22.017Z info: 	Downloaded archiveURL: https://ci.jenkins.io/job/Stapler/job/stapler/job/PR-209/1/artifact/**/*.e6cc839e83b6/*.e6cc839e83b6*/*zip*/archive.zip to path: /var/folders/7p/92qym3cj4p9bzr330tcx2m9w0000gn/T/incrementals-dQsme2/archive.zip
2021-02-04T08:38:22.018Z info: 	Downloaded file size 4173847
2021-02-04T08:38:22.022Z error: 	Invalid archive 'No applicable permissions for stapler/stapler, check jenkins-infra/repository-permissions-updater has the right configuration,'
2021-02-04T08:38:22.022Z error: 	FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?
No applicable permissions for stapler/stapler, check jenkins-infra/repository-permissions-updater has the right configuration, from https://ci.jenkins.io/job/Stapler/job/stapler/job/PR-209/1/artifact/**/*.e6cc839e83b6/*.e6cc839e83b6*/*zip*/archive.zip
    at IncrementalsPlugin.main (/Users/timja/code/jenkins/incrementals-publisher/IncrementalsPlugin.js:217:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/timja/code/jenkins/incrementals-publisher/index.js:96:13
```

</details>

<details>
<summary>New log set failing to publish to artifactory</summary>

```
2021-02-04T08:32:12.989Z info: 	Retrieving metadata from https://ci.jenkins.io/job/Stapler/job/stapler/job/PR-209/1/api/json?tree=actions[revision[hash,pullHash]]
2021-02-04T08:32:14.011Z info: 	Metadata loaded repo: stapler/stapler hash: e6cc839e83b65e27a0a860a44ae30f71fcbfb518
2021-02-04T08:32:14.013Z info: 	Prepared a temp dir for the archive /var/folders/7p/92qym3cj4p9bzr330tcx2m9w0000gn/T/incrementals-ZBNf7q
2021-02-04T08:32:16.745Z info: 	Downloaded archiveURL: https://ci.jenkins.io/job/Stapler/job/stapler/job/PR-209/1/artifact/**/*.e6cc839e83b6/*.e6cc839e83b6*/*zip*/archive.zip to path: /var/folders/7p/92qym3cj4p9bzr330tcx2m9w0000gn/T/incrementals-ZBNf7q/archive.zip
2021-02-04T08:32:16.745Z info: 	Downloaded file size 4173847
2021-02-04T08:32:16.755Z info: 	Parsed org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6.pom with url=https://github.com/stapler/stapler/stapler-groovy tag=e6cc839e83b65e27a0a860a44ae30f71fcbfb518 GAV=org.kohsuke.stapler:stapler-groovy:1.263-rc1482.e6cc839e83b6
2021-02-04T08:32:16.759Z info: 	Parsed org/kohsuke/stapler/stapler-jelly/1.263-rc1482.e6cc839e83b6/stapler-jelly-1.263-rc1482.e6cc839e83b6.pom with url=https://github.com/stapler/stapler/stapler-jelly tag=e6cc839e83b65e27a0a860a44ae30f71fcbfb518 GAV=org.kohsuke.stapler:stapler-jelly:1.263-rc1482.e6cc839e83b6
2021-02-04T08:32:16.764Z info: 	Parsed org/kohsuke/stapler/stapler-jrebel/1.263-rc1482.e6cc839e83b6/stapler-jrebel-1.263-rc1482.e6cc839e83b6.pom with url=https://github.com/stapler/stapler/stapler-jrebel tag=e6cc839e83b65e27a0a860a44ae30f71fcbfb518 GAV=org.kohsuke.stapler:stapler-jrebel:1.263-rc1482.e6cc839e83b6
2021-02-04T08:32:16.767Z info: 	Parsed org/kohsuke/stapler/stapler-jruby/1.263-rc1482.e6cc839e83b6/stapler-jruby-1.263-rc1482.e6cc839e83b6.pom with url=https://github.com/stapler/stapler/stapler-jruby tag=e6cc839e83b65e27a0a860a44ae30f71fcbfb518 GAV=org.kohsuke.stapler:stapler-jruby:1.263-rc1482.e6cc839e83b6
2021-02-04T08:32:16.768Z info: 	Parsed org/kohsuke/stapler/stapler-jsp/1.263-rc1482.e6cc839e83b6/stapler-jsp-1.263-rc1482.e6cc839e83b6.pom with url=https://github.com/stapler/stapler/stapler-jsp tag=e6cc839e83b65e27a0a860a44ae30f71fcbfb518 GAV=org.kohsuke.stapler:stapler-jsp:1.263-rc1482.e6cc839e83b6
2021-02-04T08:32:16.770Z info: 	Parsed org/kohsuke/stapler/stapler-parent/1.263-rc1482.e6cc839e83b6/stapler-parent-1.263-rc1482.e6cc839e83b6.pom with url=https://github.com/stapler/stapler tag=e6cc839e83b65e27a0a860a44ae30f71fcbfb518 GAV=org.kohsuke.stapler:stapler-parent:1.263-rc1482.e6cc839e83b6
2021-02-04T08:32:16.774Z info: 	Parsed org/kohsuke/stapler/stapler/1.263-rc1482.e6cc839e83b6/stapler-1.263-rc1482.e6cc839e83b6.pom with url=https://github.com/stapler/stapler/stapler tag=e6cc839e83b65e27a0a860a44ae30f71fcbfb518 GAV=org.kohsuke.stapler:stapler:1.263-rc1482.e6cc839e83b6
2021-02-04T08:32:16.775Z info: 	Archive entries [
  'org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6-javadoc.jar',
  'org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6-sources.jar',
  'org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6-test-sources.jar',
  'org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6.jar',
  'org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6.pom',
  'org/kohsuke/stapler/stapler-jelly/1.263-rc1482.e6cc839e83b6/stapler-jelly-1.263-rc1482.e6cc839e83b6-javadoc.jar',
  'org/kohsuke/stapler/stapler-jelly/1.263-rc1482.e6cc839e83b6/stapler-jelly-1.263-rc1482.e6cc839e83b6-sources.jar',
  'org/kohsuke/stapler/stapler-jelly/1.263-rc1482.e6cc839e83b6/stapler-jelly-1.263-rc1482.e6cc839e83b6-test-sources.jar',
  'org/kohsuke/stapler/stapler-jelly/1.263-rc1482.e6cc839e83b6/stapler-jelly-1.263-rc1482.e6cc839e83b6.jar',
  'org/kohsuke/stapler/stapler-jelly/1.263-rc1482.e6cc839e83b6/stapler-jelly-1.263-rc1482.e6cc839e83b6.pom',
  'org/kohsuke/stapler/stapler-jrebel/1.263-rc1482.e6cc839e83b6/stapler-jrebel-1.263-rc1482.e6cc839e83b6-javadoc.jar',
  'org/kohsuke/stapler/stapler-jrebel/1.263-rc1482.e6cc839e83b6/stapler-jrebel-1.263-rc1482.e6cc839e83b6-sources.jar',
  'org/kohsuke/stapler/stapler-jrebel/1.263-rc1482.e6cc839e83b6/stapler-jrebel-1.263-rc1482.e6cc839e83b6.jar',
  'org/kohsuke/stapler/stapler-jrebel/1.263-rc1482.e6cc839e83b6/stapler-jrebel-1.263-rc1482.e6cc839e83b6.pom',
  'org/kohsuke/stapler/stapler-jruby/1.263-rc1482.e6cc839e83b6/stapler-jruby-1.263-rc1482.e6cc839e83b6-javadoc.jar',
  'org/kohsuke/stapler/stapler-jruby/1.263-rc1482.e6cc839e83b6/stapler-jruby-1.263-rc1482.e6cc839e83b6-sources.jar',
  'org/kohsuke/stapler/stapler-jruby/1.263-rc1482.e6cc839e83b6/stapler-jruby-1.263-rc1482.e6cc839e83b6-test-sources.jar',
  'org/kohsuke/stapler/stapler-jruby/1.263-rc1482.e6cc839e83b6/stapler-jruby-1.263-rc1482.e6cc839e83b6.jar',
  'org/kohsuke/stapler/stapler-jruby/1.263-rc1482.e6cc839e83b6/stapler-jruby-1.263-rc1482.e6cc839e83b6.pom',
  'org/kohsuke/stapler/stapler-jsp/1.263-rc1482.e6cc839e83b6/stapler-jsp-1.263-rc1482.e6cc839e83b6-javadoc.jar',
  'org/kohsuke/stapler/stapler-jsp/1.263-rc1482.e6cc839e83b6/stapler-jsp-1.263-rc1482.e6cc839e83b6-sources.jar',
  'org/kohsuke/stapler/stapler-jsp/1.263-rc1482.e6cc839e83b6/stapler-jsp-1.263-rc1482.e6cc839e83b6.jar',
  'org/kohsuke/stapler/stapler-jsp/1.263-rc1482.e6cc839e83b6/stapler-jsp-1.263-rc1482.e6cc839e83b6.pom',
  'org/kohsuke/stapler/stapler-parent/1.263-rc1482.e6cc839e83b6/stapler-parent-1.263-rc1482.e6cc839e83b6.pom',
  'org/kohsuke/stapler/stapler/1.263-rc1482.e6cc839e83b6/stapler-1.263-rc1482.e6cc839e83b6-javadoc.jar',
  'org/kohsuke/stapler/stapler/1.263-rc1482.e6cc839e83b6/stapler-1.263-rc1482.e6cc839e83b6-sources.jar',
  'org/kohsuke/stapler/stapler/1.263-rc1482.e6cc839e83b6/stapler-1.263-rc1482.e6cc839e83b6-test-sources.jar',
  'org/kohsuke/stapler/stapler/1.263-rc1482.e6cc839e83b6/stapler-1.263-rc1482.e6cc839e83b6-tests.jar',
  'org/kohsuke/stapler/stapler/1.263-rc1482.e6cc839e83b6/stapler-1.263-rc1482.e6cc839e83b6.jar',
  'org/kohsuke/stapler/stapler/1.263-rc1482.e6cc839e83b6/stapler-1.263-rc1482.e6cc839e83b6.pom',
  [length]: 30
]
2021-02-04T08:32:16.777Z info: 	Found a POM org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6.pom
2021-02-04T08:32:21.862Z info: 	Upload result for pom: https://repo.jenkins-ci.org/incrementals/org/kohsuke/stapler/stapler-groovy/1.263-rc1482.e6cc839e83b6/stapler-groovy-1.263-rc1482.e6cc839e83b6.pom, status: 403, full error: {
  "errors" : [ {
    "status" : 403,
    "message" : "Bad props auth token: apiKey=invalid-key"
  } ]
}
2021-02-04T08:32:22.154Z error: 	Failed to create github status, code: 404 for repo: stapler/stapler, check your GitHub credentials
```

</details>

<details>
<summary>Old logs</summary>

```
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/rest-list-parameter/1.2.3-rc191.1114f1d2212b/rest-list-parameter-1.2.3-rc191.1114f1d2212b.pom from https://ci.jenkins.io/job/Plugins/job/rest-list-parameter-plugin/job/PR-43/1/artifact/**/*.1114f1d2212b/*.1114f1d2212b*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/rest-list-parameter/1.2.3-rc192.1cd3754d0efb/rest-list-parameter-1.2.3-rc192.1cd3754d0efb.pom from https://ci.jenkins.io/job/Plugins/job/rest-list-parameter-plugin/job/main/63/artifact/**/*.1cd3754d0efb/*.1cd3754d0efb*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/metrics-datadog/1.1-rc31.3e845fb3d686/metrics-datadog-1.1-rc31.3e845fb3d686.pom from https://ci.jenkins.io/job/Plugins/job/metrics-datadog-plugin/job/PR-8/1/artifact/**/*.3e845fb3d686/*.3e845fb3d686*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/lighthouse-report/1.1.1-rc94.cd7ab1cbe3d9/lighthouse-report-1.1.1-rc94.cd7ab1cbe3d9.pom from https://ci.jenkins.io/job/Plugins/job/lighthouse-report-plugin/job/PR-88/1/artifact/**/*.cd7ab1cbe3d9/*.cd7ab1cbe3d9*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/custom-folder-icon/1.5-rc60.a1ac0e089286/custom-folder-icon-1.5-rc60.a1ac0e089286.pom from https://ci.jenkins.io/job/Plugins/job/custom-folder-icon-plugin/job/PR-16/1/artifact/**/*.a1ac0e089286/*.a1ac0e089286*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/lighthouse-report/1.1.1-rc96.d0264a4f5acd/lighthouse-report-1.1.1-rc96.d0264a4f5acd.pom from https://ci.jenkins.io/job/Plugins/job/lighthouse-report-plugin/job/master/50/artifact/**/*.d0264a4f5acd/*.d0264a4f5acd*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/lighthouse-report/1.1.1-rc96.0229200cb882/lighthouse-report-1.1.1-rc96.0229200cb882.pom from https://ci.jenkins.io/job/Plugins/job/lighthouse-report-plugin/job/PR-90/1/artifact/**/*.0229200cb882/*.0229200cb882*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/lighthouse-report/1.1.1-rc98.c3f3fd51123d/lighthouse-report-1.1.1-rc98.c3f3fd51123d.pom from https://ci.jenkins.io/job/Plugins/job/lighthouse-report-plugin/job/master/51/artifact/**/*.c3f3fd51123d/*.c3f3fd51123d*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/custom-folder-icon/1.5-rc61.54120b5e2c12/custom-folder-icon-1.5-rc61.54120b5e2c12.pom from https://ci.jenkins.io/job/Plugins/job/custom-folder-icon-plugin/job/master/34/artifact/**/*.54120b5e2c12/*.54120b5e2c12*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/rest-list-parameter/1.2.3-rc193.1516683ce1a5/rest-list-parameter-1.2.3-rc193.1516683ce1a5.pom from https://ci.jenkins.io/job/Plugins/job/rest-list-parameter-plugin/job/PR-45/1/artifact/**/*.1516683ce1a5/*.1516683ce1a5*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
{"message":"FailRequestError: Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: ZIP error: Error: Missing <tag> section in <scm> of io/jenkins/plugins/rest-list-parameter/1.2.3-rc193.3ee8b9d01e11/rest-list-parameter-1.2.3-rc193.3ee8b9d01e11.pom from https://ci.jenkins.io/job/Plugins/job/rest-list-parameter-plugin/job/PR-44/1/artifact/**/*.3ee8b9d01e11/*.3ee8b9d01e11*/*zip*/archive.zip\n    at IncrementalsPlugin.main (/usr/src/app/IncrementalsPlugin.js:218:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)","level":"error"}
```

</details>

### CODE REVIEW

**Review by jglick — APPROVED** [review]
Thanks!

**Review by markyjackson-taulia — APPROVED** [review]

**Comment by timja** [comment]
either CI is dead or it doesn't run here =/

VPN is down so I can't check

**Review by basil — APPROVED** [review]

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+14.15.4
```
Comment (halkeye) [inline .nvmrc:1]: nvmrc is pretty nvm specific, so I recommend against this file. If you want to set a minimum version do >= 14 in engine in package.json

File: test/index-test.js [hunk 2]
```diff
@@ -147,7 +147,8 @@ describe('Handling incremental publisher webhook events', function() {
     it('should output an error', async function() {
       data.body.build_url = 'https://ci.jenkins.io/job/Tools/job/bom/job/PR-22/5/';
       await run();
-      assert.equal(ctx.res.body, 'Invalid archive retrieved from Jenkins, perhaps the plugin is not properly incrementalized?\nError: This is my error from https://ci.jenkins.io/job/Tools/job/bom/job/PR-22/5/artifact/**/*-rc*.55219da23b98/*-rc*.55219da23b98*/*zip*/archive.zip');
+      console.log(ctx.res.body)
```
Comment (halkeye) [inline test/index-test.js:150]:
```suggestion
```
I'm pretty sure this random console.log was left behind.

File: .nvmrc (hunk 1 above)
Comment (timja) [reply .nvmrc:1]: My problem was that I can’t get the bcrypt module to compile on latest node, so I wanted to set a known working version 

I can’t see any harm adding it

**Pull request merged**